include/envoy/http/alternate_protocols_cache.h

```diff
@@ -73,34 +79,35 @@ class AlternateProtocolsCache {
     uint32_t port_;
   };
 
-  explicit AlternateProtocolsCache(TimeSource& time_source);
-
-  // Sets the possible alternative protocols which can be used to connect to the
-  // specified origin. Expires after the specified expiration time.
-  void setAlternatives(const Origin& origin, const std::vector<AlternateProtocol>& protocols,
-                       const MonotonicTime& expiration);
-
-  // Returns the possible alternative protocols which can be used to connect to the
-  // specified origin, or nullptr if not alternatives are found. The returned pointer
-  // is owned by the AlternateProtocolsCache and is valid until the next operation on
-  // AlternateProtocolsCache.
-  OptRef<const std::vector<AlternateProtocol>> findAlternatives(const Origin& origin);
-
-  // Returns the number of entries in the map.
-  size_t size() const;
-
-private:
-  struct Entry {
-    std::vector<AlternateProtocol> protocols_;
-    MonotonicTime expiration_;
-  };
-
-  // Time source used to check expiration of entries.
-  TimeSource& time_source_;
-
-  // Map from hostname to list of alternate protocols.
-  // TODO(RyanTheOptimist): Add a limit to the size of this map and evict based on usage.
-  std::map<Origin, Entry> protocols_;
+  virtual ~AlternateProtocolsCache() = default;
+
+  /**
+   * Sets the possible alternative protocols which can be used to connect to the
+   * specified origin. Expires after the specified expiration time.
+   * @param origin The origin to set alternate protocols for.
+   * @param protocols A list of alternate protocols.
+   * @param expiration The time after which the alternatives are no longer valid.
+   */
+  virtual void setAlternatives(const Origin& origin,
+                               const std::vector<AlternateProtocol>& protocols,
+                               const MonotonicTime& expiration) PURE;
+
+  /**
+   * Returns the possible alternative protocols which can be used to connect to the
+   * specified origin, or nullptr if not alternatives are found. The returned pointer
+   * is owned by the AlternateProtocolsCacheImpl and is valid until the next operation on
+   * AlternateProtocolsCacheImpl.
+   * @param origin The origin to find alternate protocols for.
+   * @return An optional list of alternate protocols for the given origin.
+   *
```

Comment:
remove trailing whitespace?

Reply:
Thanks! Will remove in #16385

Reply:
Well, the tidy check found an issue so I fixed this too.